platform/security/certificate/src/main/java/org/codice/ddf/certificate/CertificateManager.java

```diff
@@ -0,0 +1,267 @@
+     * <p/>
+     * Purpose:  Creates a signing request with custom Issuer, Subject, and validity period data.
+     * <p/>
+     * Pre-condition: None
+     * <p/>
+     * Post-condition: Issuer information, subject information, and the encoded public key have been added to the returned signing request. This is valid for the number of days specified by numDaysValid parameter.
+     *
+     * @param issuerAttributes  String that contains a comma delimited list of certificate attributes such as CN, OU, and C for the issuer.
+     * @param subjectAttributes String that contains a comma delimited list of certificate attributes such as CN, OU, and C for the subject.
+     * @param numDaysValid      Integer representing the number of days for the certificate to be valid.
+     * @param pubKey            The PublicKey to be associated with the certificate signing request.
+     * @return X509v3CertificateBuilder signing request with the issuer, subject, and date fields set
+     * -------------------------------------------------------------------
+     */
+
+    public static X509v3CertificateBuilder createCustomX509SigningRequest(String issuerAttributes,
+            String subjectAttributes, int numDaysValid, PublicKey pubKey) {
+        X509v3CertificateBuilder certBuilder;
+        Long currentTimeMillis = System.currentTimeMillis();
+        X500Name theIssuer = new X500Name(issuerAttributes);
+        X500Name theSubject = new X500Name(subjectAttributes);
+        certBuilder = new X509v3CertificateBuilder(theIssuer, new BigInteger("2"),
+                (new Date(currentTimeMillis - MILLIS_IN_DAY)),
+                (new Date(currentTimeMillis + (MILLIS_IN_DAY * numDaysValid))), theSubject,
+                SubjectPublicKeyInfo.getInstance(pubKey.getEncoded()));
+
+        return certBuilder;
+    }
+
+    /**
+     * ---------------------------------------------------------------------
+     * Method: signX509Certificate
+     * <p/>
+     * Purpose:  Signs and returns a X509Certificate given a signing request and privateKey
+     * <p/>
+     * Pre-condition:  The X509v3CertificateBuilder has been initialized with subject, issuer, and data information.
+     * <p/>
+     * Post-condition: The X509 returned is signed using the privateKey privKey
+     *
+     * @param signingRequest an initialized signing request
+     * @param privKey        the PrivateKey used to sign the certificate
+     * @return X509Certificate that has been signed with the supplied PrivateKey
+     * -------------------------------------------------------------------
+     */
+
+    public static X509Certificate signX509CertificateSHA256RSA(
+            X509v3CertificateBuilder signingRequest, PrivateKey privKey) {
+        X509Certificate certificate = null;
+        try {
+            byte[] certBytes = signingRequest.build(new SHA256RSASigner(privKey)).getEncoded();
+            CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
```

Comment:
We should consider injecting CertificateFactory instead of using getInstance() inside the class.